### PR TITLE
feat: Add GCP blob storage connector

### PR DESCRIPTION
**Pull request by psyhedeliq**

### CODE REVIEW

File: .github/workflows/build-main.yaml [hunk 1]
```diff
@@ -38,6 +38,15 @@ jobs:
         with:
           node-version: 20
 
+      # TODO: Remove once we have a way to pass flags (-cheme http) and run it in the services section
```
Comment (obany) [inline .github/workflows/build-main.yaml:41]: Should say scheme not cheme 😄

File: .github/workflows/build-next.yaml [hunk 2]
```diff
@@ -38,6 +38,15 @@ jobs:
         with:
           node-version: 20
 
+      # TODO: Remove once we have a way to pass flags (-cheme http) and run it in the services section
```
Comment (obany) [inline .github/workflows/build-next.yaml:41]: Should say scheme not cheme 😄

File: .github/workflows/publish-next-all.yaml [hunk 3]
```diff
@@ -55,6 +56,15 @@ jobs:
           node-version: 20
           registry-url: 'https://registry.npmjs.org'
 
+      # TODO: Remove once we have a way to pass flags (-cheme http) and run it in the services section
```
Comment (obany) [inline .github/workflows/publish-next-all.yaml:59]: Should say scheme not cheme 😄

File: .github/workflows/publish-prod-all.yaml [hunk 4]
```diff
@@ -55,6 +56,15 @@ jobs:
           node-version: 20
           registry-url: 'https://registry.npmjs.org'
 
+      # TODO: Remove once we have a way to pass flags (-cheme http) and run it in the services section
```
Comment (obany) [inline .github/workflows/publish-prod-all.yaml:59]: Should say scheme not cheme 😄

File: packages/blob-storage-connector-gcp/README.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
+# TWIN Blob Storage Connector Google Cloud Storage
+
+Blob Storage connector implementation using [Google Cloud Storage](https://cloud.google.com/storage/docs/reference/libraries).
+
+## Installation
+
+```shell
+npm install @twin.org/blob-storage-connector-aws-s3
```
Comment (obany) [inline packages/blob-storage-connector-gcp/README.md:8]: Update to GCP

File: packages/blob-storage-connector-gcp/docs/changelog.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+# @twin.org/blob-storage-connector-gcp - Changelog
+
+## v0.0.1-next.4
```
Comment (obany) [inline packages/blob-storage-connector-gcp/docs/changelog.md:3]: Update version to 6

File: packages/blob-storage-connector-gcp/src/gcpBlobStorageConnector.ts [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+			this.CLASS_NAME,
+			nameof(options.config.bucketName),
+			options.config.bucketName
+		);
+
+		this._config = options.config;
+		this._storage = new Storage({
+			projectId: this._config.projectId,
+			apiEndpoint: this._config.apiEndpoint,
+			...(this._config.protocol === "http" ? { http: true } : {})
```
Comment (obany) [inline packages/blob-storage-connector-gcp/src/gcpBlobStorageConnector.ts:60]: Would this be simpler as 
```http: this._config.protocol === "http"```

File: packages/blob-storage-connector-gcp/src/gcpBlobStorageConnector.ts [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+	public async set(blob: Uint8Array): Promise<string> {
+		Guards.uint8Array(this.CLASS_NAME, nameof(blob), blob);
+
+		try {
+			const id = Converter.bytesToHex(Sha256.sum256(blob));
+			const bucket = this._storage.bucket(this._config.bucketName);
+			const file = bucket.file(id);
+
+			await file.save(blob, {
+				contentType: "application/octet-stream"
```
Comment (obany) [inline packages/blob-storage-connector-gcp/src/gcpBlobStorageConnector.ts:78]: We have some predefined type like `application/octet-stream` in `@twin.org/web`  MimeTypes https://github.com/twinfoundation/framework/blob/a7ca106365da663a115e476327bb7bda21350bb6/packages/web/src/models/mimeTypes.ts#L42

**Review by obany — CHANGES_REQUESTED** [review]
A few minor comments

File: packages/blob-storage-connector-gcp/src/gcpBlobStorageConnector.ts (hunk 7 above)
Comment (psyhedeliq) [reply packages/blob-storage-connector-gcp/src/gcpBlobStorageConnector.ts:60]: I tried that and I am getting this:
`Object literal may only specify known properties, and 'http' does not exist in type 'StorageOptions'.`

Comment (psyhedeliq) [reply packages/blob-storage-connector-gcp/src/gcpBlobStorageConnector.ts:60]: Should I leave it like this?

Comment (psyhedeliq) [reply packages/blob-storage-connector-gcp/src/gcpBlobStorageConnector.ts]: I removed it! I remember I added that when I was having issues with the HTTP/HTTPS, and I actually didn't need it.

**Pull request merged**